src/components/scores.jsx

```diff
@@ -0,0 +1,919 @@
+							return false
+					}
+
+				default:
+					console.warn('Invalid postMessage source or no source provided')
+					return false
+			}
+		} else {
+			throw new Error(`Error, cross domain restricted for ${origin}`)
+		}
+	}
+
+  // broadcasts a postMessage to inform Obojobo, or other platforms, about a score event
+	// Bypasses the LTI interface and provides an alternative for platforms that use embedded Materia to listen for a score
+	const sendPostMessage = (score) => {
+    if (parent.postMessage && JSON.stringify) {
+			parent.postMessage(
+				JSON.stringify({
+					type: 'materiaScoreRecorded',
+					source: 'score-controller',
+					widget: instance,
+					score,
+				}),
+				'*'
+			)
+		}
+	}
+
+	const getAttemptNumberFromHash = () => {
+		const match = window.location.hash.match(/^#attempt-(\d+)/)
+		if (match && match[1] != null && !isNaN(match[1])) {
+			return match[1]
+		}
+		return attempts.length
+	}
+
+	const _sendToWidget = (type, args) => {
+		return scoreWidgetRef.current.contentWindow.postMessage(
+			JSON.stringify({ type, data: args }),
+			window.STATIC_CROSSDOMAIN
+		)
+	}
+
+	const _sendWidgetInit = () => {
+    if (customScoreScreen.qset == null || scoreWidgetRef.current == null) {
+      // Custom score screen failed to load, load default overview instead
+      setCustomScoreScreen({...customScoreScreen, loading: true, show: false})
+      setShowResultsTable(true)
+      setShowScoresOverview(true)
+			setInvalid(true)
+			return
+		}
+		setShow(true)
+		_sendToWidget('initWidget', [customScoreScreen.qset, customScoreScreen.scoreTable, instance, isPreview, window.MEDIA_URL])
+	}
+
+	const _sendWidgetUpdate = () => {
+		_sendToWidget('updateWidget', [qset, scoreTable])
+	}
+
+	const _setHeight = (h) => {
+		const min_h = instance.widget.height
+		let desiredHeight = Math.max(h, min_h)
+		scoreWidgetRef.current.style.height = `${desiredHeight}px`
+	}
+
+  const waitForScoreSummary = async () => {
+    while (scoreSummaryIsLoading || !scoreSummary) {
+      await new Promise(resolve => setTimeout(resolve, 500))
+    }
+  }
+
+  const prevMouseOver = () => {
+    setPrevAttemptClass('open')
+  }
+
+  const prevMouseOut = () => {
+    setPrevAttemptClass('')
+  }
+
+  const attemptClick = () => {
+    if (isMobile.any()) {
+			setPrevAttemptClass('')
+		}
+  }
+
+  let previousAttempts = null
+  if (!hidePreviousAttempts && !isPreview && !guestAccess) {
+    previousAttempts = (
+      <nav className={`header-element previous-attempts ${prevAttemptClass}`} onMouseOver={prevMouseOver} onMouseOut={prevMouseOut}>
+        <h1 onClick={prevMouseOver}>Prev. Attempts</h1>
+        <ul onMouseOver={prevMouseOver}>
+          {attempts.forEach((attempt, index) => (
+            <li key={index}>
+              <a href={`#attempt-${index}`} onClick={attemptClick}>Attempt {index}: <span className="score">{ attempt.roundedPercent}%</span><span className="date">{ dates[index]}</span></a>
```

Comment:
`dates[index]` is undefined on this line. I think you meant `attemptDates[index]`